### ISSUE TITLE
Add option to lock out by ip OR username

### ISSUE THREAD

**Issue by PetrDlouhy** [jazzband/django-axes#608]
Maybe I did get something wrong, but I think, that it would be possible to set increase safety options of `django-axes`.
With the default option, the attacker can have access to large stack of IP addresses (i.e. through IPv6, darkweb or botnet), and with that he can make N login attempts per IP, which might be sufficient to guess someones password.

On the other hand the other options are insufficient either (at least what I found). The `AXES_LOCK_OUT_BY_COMBINATION_USER_AND_IP` has same problem as the previous case and the `AXES_ONLY_USER_FAILURES` would allow many N*U login attempts per username to attacker with username list of the server (and he could repeat his attack after cooloff time or filling in captcha depending on server configuration).

I think, that `django-axes` needs to (at least offer, but I think it is good default) locking out based on username OR ip address attempt count. With that I mean, that after N attempts from IP address the access from that address is locked out regardless of username and after N attempts with the same username, the username is locked out regardless of IP address.

**Comment by aleksihakli** [comment]
Yeah, it would be good to offer a combination of both of these options. The project is open for PRs.

A few notes on the implementation of this feature:

1. new unit tests need to be implemented for this feature,
2. existing handlers need to be refactored in order to facilitate this, and
3. it could be good to refactor the existing test base while implementing this since it has a lot of old cruft.
